core/server/api/schedules.js

```diff
@@ -0,0 +1,61 @@
+var _ = require('lodash'),
+    Promise = require('bluebird'),
+    moment = require('moment'),
+    config = require('../config'),
+    pipeline = require(config.paths.corePath + '/server/utils/pipeline'),
+    dataProvider = require(config.paths.corePath + '/server/models'),
+    i18n = require(config.paths.corePath + '/server/i18n'),
+    errors = require(config.paths.corePath + '/server/errors'),
+    utils = require('./utils');
+
+/**
+ * publish a scheduled post
+ */
+exports.publishPost = function publishPost(options) {
+    options = options || {};
+
+    // CASE: only the scheduler client is allowed to publish (hardcoded because of missing client permission system)
+    if (!options.context || !options.context.client || options.context.client !== 'ghost-scheduler') {
+        return Promise.reject(new errors.NoPermissionError(i18n.t('errors.permissions.noPermissionToAction')));
+    }
+
+    return pipeline([
+        utils.validate('posts', {opts: utils.idDefaultOptions}),
+        function (cleanOptions) {
+            return dataProvider.Post.edit({status: 'published'}, _.omit(cleanOptions, 'data'))
```

Comment:
X-Cache-Invalidation headers are what should be used by the scheduler to determine the `clearCache` property?
Also there is no handling here for a 404 for this endpoint - it will only 404 if the post is not found, not if the task is not found. 

We have to model-representation of a task but we still have logic for it. The task exists if the post is set to scheduled and the published_at date is in the past or near future.